cpp/src/arrow/sparse_tensor.cc

```diff
@@ -33,7 +33,7 @@ namespace arrow {
 // SparseIndex
 
 Status SparseIndex::ValidateShape(const std::vector<int64_t>& shape) const {
-  if (!std::all_of(shape.begin(), shape.end(), [](int64_t x) { return x > 0; })) {
+  if (!std::all_of(shape.begin(), shape.end(), [](int64_t x) { return x >= 0; })) {
```

Comment:
This and the same change in tensor.c below is necessary to accept making tensors with 0-length dimensions at https://github.com/apache/arrow/blob/master/cpp/src/arrow/ipc/read_write_test.cc#L1112.